### ISSUE TITLE
Add subscription to SharedImageGalleryDestination object

### ISSUE THREAD

**Issue by DOMZE** [hashicorp/packer#9790]
#### Community Note

Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request.
If you are interested in working on this issue or have submitted a pull request, please leave a comment.

#### Description

In the [SharedImageGalleryDestination](https://github.com/hashicorp/packer/blob/master/builder/azure/chroot/shared_image_gallery_destination.go) object, it would be nice to have the subscription as an option. The current configuration falls back to the current subscription packer is running in. It would be the same idea as the "source" (i.e. SharedImageGallery) where it can be filled in.

If the option is not provided, the fallback would be to use the current subscriptionId

#### Use Case(s)

Individuals/companies may have a central SIG that they use for all their provisioning

#### Potential configuration

```
"shared_image_gallery_destination": {
    "subscription": "00000000-0000-0000-0000-00000000000",
    "resource_group": "ResourceGroup",
    "gallery_name": "GalleryName",
    "image_name": "ImageName",
    "image_version": "1.0.0",
    "replication_regions": ["regionA", "regionB", "regionC"]
}
```

**Comment by sylviamoss** [comment]
Hey @DOMZE, thanks for reaching out! We'll take a look when we get a chance. I think it makes sense and looking at the code this seems to be possible.

**Comment by nywilken** [comment]
Hi :wave: we have an open issue for this request  https://github.com/hashicorp/packer/issues/8608. I’m going to close this issue and mark the original issue for the 1.6.3 milestone. Please add any additional comments on the original issue.  If you think this is different from 8608 please let us know and we will gladly reopen. Cheers!

**Comment by nywilken** [comment]
Hello :wave: thanks again for reporting, this should be fixed by #9875 ; some test binaries are available via the link below if you would like to test and confirm that the feature works as expected. Thanks!

https://app.circleci.com/pipelines/github/hashicorp/packer/6995/workflows/b846f4fe-7a13-4789-95d5-23a0e3b11c35/jobs/78838/artifacts
